operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -158,26 +185,50 @@ public static File createFileStore(String prefix, String suffix, byte[] bytes) {
      * Decode binary item from Kubernetes Secret from base64 into byte array
      *
      * @param secret    Kubernetes Secret
-     * @param key       Key which should be retrieved and decoded
+     * @param field     Field which should be retrieved and decoded
      * @return          Decoded bytes
      */
-    public static byte[] decodeFromSecret(Secret secret, String key) {
-        return Base64.getDecoder().decode(secret.getData().get(key));
+    public static byte[] decodeBase64FieldFromSecret(Secret secret, String field) {
+        return Optional.ofNullable(secret)
+                .map(Secret::getData)
+                .map(data -> data.get(field))
+                .map(value -> Base64.getDecoder().decode(value))
+                .orElseThrow(() -> {
+                    String name = Optional.ofNullable(secret)
+                            .map(Secret::getMetadata)
+                            .map(ObjectMeta::getName)
+                            .orElse("unknown");
```

Comment:
When would it not have a name?

Reply:
If the secret passed into the method is null

Reply:
I guess it would be better to check that separately and possibly throw a different error and not mask it behind some `unknown` name?

Reply:
Maybe it suggests something wrong in the design -> you do not know if the secret is required or not at the point where you process it and that forces you to do this kind of weird pre-processing?

Reply:
I've updated it to do two steps, so it first throws an exception if the secret is null and then if it fails to get the field it can log knowing the secret name and namespace must be present.